src/hooks/useQueryParams.tsx

```diff
@@ -44,9 +45,9 @@ const useQueryParams = (
   defaultValues: QueryParams,
   validKeys?: string[],
 ): UseQueryParamsOutput => {
-  const history = useHistory();
+  const location = useLocation();
```

Comment:
Curious: What necessitated this change?

Reply:
Gatsby uses Reach Router (`@reach/router`) instead of React Router (`react-router`), which has a very similar but different API. Here is the complete set up substitutions I made:
- `useLocation()` => `useLocation()` (no change, they both have `useLocation()`)
- `useHistory().location` => `useLocation()`
- `useHistory().push(...)` => `navigate()`
- `useHistory().replace(...)` => `navigate(..., {replace: true})`

We can't mix and match routing libraries because each library looks for its own provider.

Note: React Router's API will eventually look more like the Reach Router one.
https://reacttraining.com/blog/reach-react-router-future/

Reply:
Ah I had no idea that was a 3rd party library. Given all of this material, I feel better about this!